src/nvim/channel.h

```diff
@@ -77,7 +77,8 @@ struct Channel {
   CallbackReader on_stderr;
   Callback on_exit;
 
-  varnumber_T *status_ptr; // TODO: refactor?
+  // Exit status. Note: may be overwritten by f_jobwait.
+  varnumber_T status;
```

Comment:
Is this member needed in the channel struct? Only channels that use `proc` have a status and that status could be directly read from `channel->stream.proc->status`. It looks like that this variable is only here to simplify `f_jobwait()` a bit.